### PR TITLE
Edit action item assignee

### PR DESCRIPTION
**Pull request by samdec11**
__Description of Change(s) Introduced:__ Facilitators now have the ability to edit the assignee of an action item.

__Description of Testing Applied:__ Frontend component tests and backend to ensure expected messages are received and handled accordingly, changing the assigned assignee_id

__Relevant Screenshots/GIFs:__ (if applicable)

screen shot 2018-01-12 at 10 58 04 am

__Relevant github Issue:__ (if applicable)

- #350

### CODE REVIEW

File: test/components/idea_edit_form_test.js [hunk 1]
```diff
@@ -3,17 +3,27 @@ import { shallow } from "enzyme"
...
 
 import IdeaEditForm from "../../web/static/js/components/idea_edit_form"
+import STAGES from "../../web/static/js/configs/stages"
+
+const { IDEA_GENERATION, ACTION_ITEMS } = STAGES
 
 describe("<IdeaEditForm />", () => {
   const idea = { id: 999, body: "redundant tests", userId: 1 }
-  const stage = "idea-generation"
+  const stage = IDEA_GENERATION
```
Comment (vanderhoop) [inline test/components/idea_edit_form_test.js]: 💯  ❤️  💯

File: web/models/idea.ex [hunk 2]
```diff
@@ -15,10 +15,11 @@ defmodule RemoteRetro.Idea do
   end
 
   @required_fields [:category, :body, :retro_id, :user_id]
+  @mutable_fields [:assignee_id | @required_fields]
 
   def changeset(struct, params \\ %{}) do
     struct
-    |> cast(params, @required_fields)
+    |> cast(params, @mutable_fields)
```
Comment (vanderhoop) [inline web/models/idea.ex]: beautiful

File: web/channels/retro_channel.ex [hunk 3]
```diff
@@ -109,7 +109,7 @@ defmodule RemoteRetro.RetroChannel do
     {:noreply, socket}
   end
 
-  defp add_idea!(%{"body" => body, "category" => category, "userId" => user_id, "assigneeId" => assignee_id}, socket) do
+  defp add_idea!(%{"body" => body, "category" => category, "userId" => user_id, "assignee_id" => assignee_id}, socket) do
```
Comment (vanderhoop) [inline web/channels/retro_channel.ex]: why this change? not vehemently opposed to it, but it does seem inconsistent with the camelCased `userId` coming from the front end

File: web/static/js/components/idea_edit_form.jsx [hunk 4]
```diff
@@ -9,7 +9,9 @@ class IdeaEditForm extends Component {
     this.state = {
       ideaBody: props.idea.body,
       ideaCategory: props.idea.category,
+      assignee_id: props.idea.assignee_id,
```
Comment (vanderhoop) [inline web/static/js/components/idea_edit_form.jsx]: here, we could keep the state properties consistent by adding the `idea` prefix

File: web/static/js/components/idea_edit_form.jsx [hunk 5]
```diff
@@ -63,6 +71,16 @@ class IdeaEditForm extends Component {
             <option key={category} value={category}>{category}</option>
           ))}
         </select>}
+        {stage === "action-items" && <select
```
Comment (vanderhoop) [inline web/static/js/components/idea_edit_form.jsx]: a couple things here. The first is that we should attempt to re-use the `SelectDropdown` component that we introduced in the "action items assignment" story. The second is that we should leverage the stage constants found in the configs/stages.js file, both here (line 74) and [up above](https://github.com/stride-nyc/remote_retro/blob/96139549351ce2c9b93ced38eedf36c63e4ef683/web/static/js/components/idea_edit_form.jsx#L65)

File: web/static/js/components/idea_submission_form.jsx [hunk 6]
```diff
@@ -49,7 +49,7 @@ export class IdeaSubmissionForm extends Component {
     const { currentUser } = this.props
     event.preventDefault()
     const newIdea = { ...this.state, userId: currentUser.id }
-    if (!newIdea.assigneeId) { newIdea.assigneeId = null }
```
Comment (vanderhoop) [inline web/static/js/components/idea_submission_form.jsx]: rather than doing this assignment just before submission, we can just set assigneeId to `null` where we declare `this.state = { ... }` in the constructor

File: web/static/js/components/idea_submission_form.jsx [hunk 7]
```diff
@@ -65,12 +65,12 @@ export class IdeaSubmissionForm extends Component {
...
 
   handleAssigneeChange(event) {
-    this.setState({ assigneeId: Number.parseInt(event.target.value, 10) })
+    this.setState({ assignee_id: Number.parseInt(event.target.value, 10) })
   }
 
   render() {
     const { users, stage } = this.props
-    const { assigneeId, body, ideaEntryStarted, category } = this.state
+    const { assignee_id: assigneeId, body, ideaEntryStarted, category } = this.state
```
Comment (vanderhoop) [inline web/static/js/components/idea_submission_form.jsx]: the more i see it, the more the change to a snake_cased `assignee_id` (client-side, at least) seems to complicate things

File: web/static/js/components/idea_edit_form.jsx (hunk 5 above)
Comment (vanderhoop) [reply web/static/js/components/idea_edit_form.jsx]: and in fact, we should also use the `SelectDropdown` [up above as well](https://github.com/stride-nyc/remote_retro/blob/96139549351ce2c9b93ced38eedf36c63e4ef683/web/static/js/components/idea_edit_form.jsx#L64-L73).

File: web/static/js/components/idea_submission_form.jsx (hunk 6 above)
Comment (samdec11) [reply web/static/js/components/idea_submission_form.jsx]: Done

**Comment by vanderhoop** [comment]
hey @samdec11, changes are looking good! Just need the `SelectDropdown` to take the pre-populated value. As right now, the "edit assignee" dropdown's initial value doesn't show anything. Fix that and we're golden!

screen shot 2018-01-14 at 9 44 19 am

File: web/static/js/components/idea_submission_form.jsx [hunk 8]
```diff
@@ -101,7 +101,7 @@ export class IdeaSubmissionForm extends Component {
       pointerText = !ideaEntryStarted ? "Create Action Items!" : ""
       dropdownProps = {
         labelName: "assignee",
-        value: assigneeId,
+        value: assigneeId || "",
```
Comment (vanderhoop) [inline web/static/js/components/idea_submission_form.jsx]: maybe i'm being paranoid, but does this fallback value throw errors server side? i think we'll want assigneeId to explicitly be passed as `null` when passed to the server in cases where there is no assigneeId

Comment (samdec11) [reply web/static/js/components/idea_submission_form.jsx]: It doesn't send the fallback to the server. This is only there to stop console errors resulting from having the initial value from the state set as `null`.
